File: backend/SportEventManager/SportEventManager.Domain/Entities/Event.cs
using SportEventManager.Domain.Common;

namespace SportEventManager.Domain.Entities
{
    public class Event : BaseEntity
    {
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public int? VenueId { get; set; }
 
        public string Status { get; set; } = "Scheduled";
        public int? MaxParticipants { get; set; }

        public Venue Venue { get; set; }
        public ICollection<EventSchedule> Schedules { get; set; }
    }
}

File: src/types/event.ts
export interface Event {
  id: number;
  title: string;
  description: string;
  sportId: number;
  startDate: string;
  endDate: string;
  location: string;
  status: EventStatus;
  maxParticipants: number;
  currentParticipants: number;
  organizerId: number;
  createdAt: string;
  updatedAt: string;
}

export type EventStatus =
  | "scheduled"
  | "in_progress"
  | "completed"
  | "cancelled";

export interface EventFilters {
  search?: string;
  sportId?: number;
  status?: EventStatus;
  startDate?: string;
  endDate?: string;
}

export interface EventFormData {
  title: string;
  description: string;
  sportId: number;
  startDate: string;
  endDate: string;
  location: string;
  maxParticipants: number;
  status: EventStatus;
}
